lib/IRGen/GenMeta.cpp

```diff
@@ -991,43 +988,26 @@ namespace {
                                                               NotForDefinition);
       B.addRelativeAddress(cache);
 
-      // Relative pointer to the metadata.
-      auto type = Type->getDeclaredTypeInContext()->getCanonicalType();
-      auto metadata = IGM.getAddrOfTypeMetadata(type);
-      B.addRelativeAddress(metadata);
+      asImpl().addIncompleteMetadataOrRelocationFunction();
 
       // Completion function.
       auto completionFunction =
         IGM.getAddrOfTypeMetadataCompletionFunction(Type, NotForDefinition);
       B.addRelativeAddress(completionFunction);
     }
 
+    void addIncompleteMetadataOrRelocationFunction() {
+      // Relative pointer to the metadata.
+      auto type = Type->getDeclaredTypeInContext()->getCanonicalType();
+      auto metadata = IGM.getAddrOfTypeMetadata(type);
+      B.addRelativeAddress(metadata);
+    }
```

Comment:
I feel like this function ought to be called `addIncompleteMetadata`.  You can still have an `addIncompleteMetadataOrRelocationFunction` that just calls it if that's the right design.  But calling this the right thing will make the super-call to it much clearer.

Reply:
Sounds good